src/dmd/dsymbolsem.d

```diff
@@ -4764,6 +4764,29 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
         {
             auto s = (*cldec.members)[i];
             s.dsymbolSemantic(sc2);
+
+            if (cldec.classKind == ClassKind.cpp)
+            {
+                const DtorDeclaration dtor = s.isDtorDeclaration();
+                if (dtor && !dtor.isFinal())
```

Comment:
A final function can still override a base class implementation.
It seems you cannot override non-virtual destructors anyway, so it should also check for the existence of a base class destructor?